todos-app/backend/task_service/app/controllers/tasks_controller.rb

```diff
@@ -41,6 +42,12 @@ def destroy
     @task.destroy
   end
 
+  def destroy_all
+    params.require(:user_id)
+    @tasks = Task.where(user_id: @user_id)
+    @tasks.destroy_all
+  end
```

Comment:
That's a very true! In fact, this is an inherent issue with using microservice architecture. Because each service has its own dedicated DB isolated from each other, we no longer can use joins in the tables. Task data is stored in task service DB and user data is stored in user service DB. 

My implementation is the most simple but naiive and optimistic one because it assumes that every request will be successful. If it was a production grade app, I'd probably go with an event driven design like this;
![image](https://user-images.githubusercontent.com/95325950/149264964-926b3b1b-03a2-4323-9ec3-58023ff0db88.png)

But I thought this is too much for a training project 😅

Reply:
ahh... then dependent won't work for this!
Okay just leave it as it is. 😆